### PR TITLE
Add cache directory if doesnt exist

### PR DESCRIPTION
**Pull request by visiover**

### CODE REVIEW

**Comment by mindplay-dk** [comment]
Thanks, Maher :-)

There is one minor issue, in `FileCache`:

        if (! file_exists($cache_path)) {
            @mkdir($cache_path, 0777, true); // ensure that the parent path exists
        }

This will auto-create directories any number of levels deep - all the way from the root of the filesystem, which I don't think it safe? Bad configuration could lead to entire directory-structures getting accidentally created.

I think we should allow auto-creation of the immediate parent folder only, e.g.:

        if (! file_exists($cache_path) && file_exists(dirname($cache_path))) {
            @mkdir($cache_path, 0777); // ensure that the parent path exists
        }

This work for most use-cases - as long as you're pointing to the correct run-time folder, a sub-folder under it will be created.

**Comment by visiover** [comment]
@mindplay-dk 

Good point. Auto-creation of the immediate parent folder only should be allowed to avoid any wrong directory-structure.

Will add it right now

File: src/FileCache.php [hunk 1]
```diff
@@ -46,6 +46,10 @@ class FileCache implements CacheInterface, CounterInterface
      */
     public function __construct($cache_path, $default_ttl)
     {
+        if (! file_exists($cache_path) && file_exists(dirname($cache_path))) {
+            @mkdir($cache_path, 0777, true); // ensure that the parent path exists
```
Comment (mindplay-dk) [inline src/FileCache.php]: Minor detail, but we don't need the `true` argument anymore, do we?

Comment (visiover) [reply src/FileCache.php]: no i don't think that we need to the creation of nested directories. 
we just need to create the cache parent folder.

**Comment by mindplay-dk** [comment]
Merged and released as `0.1.1` - thanks :-)

**Pull request merged**